### ISSUE TITLE
Clippy fix makes breaking change on warning: usage of `contains_key` followed by `insert` on a `HashMap`

### ISSUE THREAD

**Issue by dchen5022** [rust-lang/rust-clippy#13649]
### Summary

Clippy attempts to fix the following diagnostic:
warning: usage of `contains_key` followed by `insert` on a `HashMap`
   --> src/environment/mod.rs:289:13
    |
289 | /             if relay_drivers.contains_key(&name) {
290 | |                 res.error(format!("Duplicate relay driver {} in drivers file.", name));
291 | |             } else {
292 | |                 relay_drivers.insert(name, relay_driver);
293 | |             }
    | |_____________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_entry
help: try
    |
289 ~             if let std::collections::hash_map::Entry::Vacant(e) = relay_drivers.entry(name) {
290 +                 e.insert(relay_driver);
291 +             } else {
292 +                 res.error(format!("Duplicate relay driver {} in drivers file.", name));
293 +             }
    |

Which results in `name` being borrowed after moved in the first line.

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: aarch64-apple-darwin
release: 1.81.0
LLVM version: 18.1.7

**Comment by dchen5022** [comment]
Basically what this person said on another issue: https://github.com/rust-lang/rust-clippy/issues/1450#issuecomment-460984447

**Comment by zjp-CN** [comment]
Seems to have been fixed since https://github.com/rust-lang/rust-clippy/pull/14314 .

**Comment by ada4a** [comment]
Since https://github.com/rust-lang/rust-clippy/pull/15546, this no longer creates an invalid suggestion at all; the diagnostic now looks like this:
```
warning: usage of `contains_key` followed by `insert` on a `HashMap`
  --> src/main.rs:11:5
   |
11 | /     if map.contains_key(&key) {
12 | |         error(format!("Duplicate sensor driver {} in drivers file.", key));
13 | |     } else {
14 | |         map.insert(key, val);
15 | |     }
   | |_____^
   |
   = help: consider using the `Entry` API: https://doc.rust-lang.org/std/collections/struct.HashMap.html#entry-api
```

**Comment by ada4a** [comment]
Oh and it turns out that you can indeed use the `Entry` API here:
```rs
let name = format!("{}/{}", sensor_driver.namespace, sensor_driver.function);
match sensor_drivers.entry(name) {
    Entry::Occupied(o) => {
        res.error(format!(
            "Duplicate sensor driver {} in drivers file.",
            o.key()
        ));
    }
    Entry::Vacant(v) => {
        v.insert(sensor_driver);
    }
}
```
~Maybe I'll be able to create an automated suggestion for that...~ Turned out to be rather complicated
